src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets

```diff
@@ -25,7 +25,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <ContainerName Condition="'$(ContainerName)' == ''">$(TestProduct)-$(Branch)-$(BuildMoniker)$(CloudTestContainerSuffix)</ContainerName>
+    <ContainerName>build-$([System.Guid]::NewGuid().ToString("N"))</ContainerName>
```

Comment:
This as far as I know can be overridden with a /p:Container even without the condition as it is a property group outside any Target.